### PR TITLE
Implement Unit Tests for services

### PR DESCRIPTION
**Pull request by JoyfulReaper**
Implement Unit Tests for services

### CODE REVIEW

File: CRUDRecipeEF.BL.DL/Services/IngredientService.cs [hunk 1]
```diff
@@ -22,7 +22,12 @@ public IngredientService(RecipeContext context, IMapper mapper)
         private async Task<Ingredient> GetIngredientByNameIfExists(string name)
         {
             var ingredient = await _context.Ingredients.FirstOrDefaultAsync(i => i.Name.ToLower() == name.ToLower());
-            return ingredient ?? throw new KeyNotFoundException("Ingredient doesnt exist");
+            if (ingredient == null)
+            {
+                throw new KeyNotFoundException("Ingredient doesnt exist");
```
Comment (chrisK00) [inline CRUDRecipeEF.BL.DL/Services/IngredientService.cs:27]: How come you changed this?

File: CRUDRecipeTests/Services/SQLiteIngredientServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                Assert.Equal("Apple", ingredient.Name);
+            }
+        }
+
+        [Fact]
+        public async Task Test_GetAllIngredients()
+        {
+            using (var context = new RecipeContext(ContextOptions))
+            {
+                var ingredientService = new IngredientService(context, new Mapper(autoMapperConfig));
```
Comment (chrisK00) [inline CRUDRecipeTests/Services/SQLiteIngredientServiceTests.cs:54]: Instead of making a new mapper, can we create it in the ctor and reuse the field?

**Review by chrisK00 — APPROVED** [review]
wow very well done! some really cool stuff going on 👍 thank you so much for putting down the time. its perfect

File: CRUDRecipeEF.BL.DL/Services/IngredientService.cs (hunk 1 above)
Comment (JoyfulReaper) [reply CRUDRecipeEF.BL.DL/Services/IngredientService.cs:27]: Way easier to see what is going on when running the code in the debugger, that's all :D

File: CRUDRecipeTests/Services/SQLiteIngredientServiceTests.cs (hunk 2 above)
Comment (JoyfulReaper) [reply CRUDRecipeTests/Services/SQLiteIngredientServiceTests.cs:54]: On another branch now, will take a look in a moment, but probably

Comment (JoyfulReaper) [reply CRUDRecipeTests/Services/SQLiteIngredientServiceTests.cs]: Done, good to merge now :D

**Pull request merged**